framework/sysdeps/linux/cpu_affinity.cpp

```diff
@@ -22,9 +21,22 @@ static void set_thread_name(const char *thread_name)
 
 LogicalProcessorSet ambient_logical_processor_set()
 {
-    LogicalProcessorSet result;
-    if (sched_getaffinity(0, sizeof(result.array), reinterpret_cast<cpu_set_t *>(result.array)) != 0)
-        result.clear();
+    int size = CPU_SETSIZE;
+    LogicalProcessorSet result(size);
+    int r = sched_getaffinity(0, result.size_bytes(), reinterpret_cast<cpu_set_t *>(result.array.data()));
+
+    while (r != 0 && errno == EINVAL) {
+        // increase the set size until it works
```

Comment:
typo: "while it works" or "until it does not work"

Reply:
No, this is correct: we're failing, so we increase until it works. But I'll rephrase.